processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/DefaultTaskExtractor.java

```diff
@@ -31,8 +31,13 @@
 
 @RequiredArgsConstructor
 public class DefaultTaskExtractor<T> implements TaskExtractor<T> {
+    private static final ThreadLocal<Boolean> parseAsLegacyWhenHeaderMissing = ThreadLocal.withInitial(() -> false);
```

Comment:
h-m this might be a bit too hacky, given that we would have to maintain this compatibility mode for certain period of time isn't it?

Can we do it like;

1. delay `DefaultTaskExctractor` instantiation to the time calling `ProcessorsBuilder#build`
2. so that we can instantiate it in `SubscriptionBuilder#build`
3. at that point we already have access to properties, so we can simply pass a boolean flag to the `DefaultTaskExtractor` constructor?

Reply:
Ah, that's way better. Let me fix